docs/GetStarted/getting_started_macos_bazel.md

```diff
@@ -0,0 +1,164 @@
+
+# Add to path
+$ echo -e '\n# Add bazel to PATH:\nexport PATH="${PATH}:${HOME}/bin"' >> ~/.zshrc
+$ source ~/.zshrc
+
+# Confirm the version
+$ bazel --version # 2.1.0
+```
+
+See the
+[official docs](https://docs.bazel.build/versions/master/install-os-x.html#install-with-installer-mac-os-x)
+for additional guidance if needed.
+
+### Install a Compiler
+
+We recommend Clang. GCC is not fully supported. Appropriate versions of `clang`
+and `clang++` ship with MacOS by default.
+
+Set environment variables for Bazel:
+
+```shell
+export CC=clang
+export CXX=clang++
+```
+
+### Install python3 numpy
+
+```
+$ python3 -m pip install numpy --user
+```
+
+## Clone and Build
+
+### Clone
+
+Clone the repository, initialize its submodules and configure:
+
+```shell
+$ git clone https://github.com/google/iree.git
+$ cd iree
+$ git submodule update --init
+$ python3 configure_bazel.py
+```
+
+> Tip:<br>
+> &nbsp;&nbsp;&nbsp;&nbsp;Editors and other programs can also clone the
+> repository, just make sure that they initialize the submodules.
+
+### Build
+
+Run all core tests that pass on our OSS CI:
+
+```shell
+$ bazel test -k //iree/... \
+    --test_env=IREE_VULKAN_DISABLE=1 \
+    --build_tag_filters="-nokokoro" \
+    --test_tag_filters="--nokokoro,-driver=vulkan"
+```
+
+> Tip:<br>
+> &nbsp;&nbsp;&nbsp;&nbsp;Not all tests are passing on MacOS, but the build does
+> complete successfully at the time of writing.
+
+In general, build artifacts will be under the `bazel-bin` directory at the top
+level.
+
+## Recommended user.bazelrc
+
+You can put a user.bazelrc at the root of the repository and it will be ignored
+by git. The recommended contents for Linux/MacOS are:
+
+```
+build --disk_cache=/tmp/bazel-cache
+
+# Use --config=debug to compile iree and llvm without optimizations
+# and with assertions enabled.
+build:debug --config=asserts --compilation_mode=opt '--per_file_copt=iree|llvm@-O0' --strip=never
+
+# Use --config=asserts to enable assertions in iree and llvm.
+build:asserts --compilation_mode=opt '--per_file_copt=iree|llvm@-UNDEBUG'
+```
+
+## What's next?
+
+### Take a Look Around
+
+Build all of IREE's 'tools' directory:
+
+```shell
+$ bazel build iree/tools/...
+```
+
+Check out what was built:
+
+```shell
+$ ls bazel-bin/iree/tools/
+$ ./bazel-bin/iree/tools/iree-translate --help
+```
+
+Translate a
+[MLIR file](https://github.com/google/iree/blob/master/iree/tools/test/simple.mlir)
+and execute a function in the compiled module:
+
+```shell
+$ ./bazel-bin/iree/tools/iree-run-mlir ./iree/tools/test/simple.mlir -input-value="i32=-2" -iree-hal-target-backends=vmla -print-mlir
+```
+
+### Further Reading
+
+*   For an introduction to IREE's project structure and developer tools, see
+    [Developer Overview](../developer_overview.md)
+<!--
+TODO: Link to MacOS versions of these guides once they are developed.
+*   To target GPUs using Vulkan, see
+    [Getting Started on Linux with Vulkan](getting_started_linux_vulkan.md)
```

Comment:
FYI, Vulkan on macOS could be possible through MoltenVK, but we're also thinking about adding a native Metal backend.

Reply:
Interesting!